akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorRefIgnoreSpec.scala

```diff
@@ -0,0 +1,162 @@
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.concurrent.PatienceConfiguration.{ Timeout => PatienceTimeout }
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.typed.scaladsl.adapter._
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.actor.testkit.typed.scaladsl.TestProbe
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+class ActorRefIgnoreSpec extends AnyWordSpec with ScalaFutures with Matchers with BeforeAndAfterAll with Eventually {
+
+  val testKit = ActorTestKit()
+  override def afterAll(): Unit = testKit.shutdownTestKit()
+
+  case class Request(replyTo: ActorRef[Int])
+
+  // this Actor behavior receives simple request and answers back total number of
+  // messages it received so far
+  val askMeActorBehavior: Behavior[Request] = {
+    def internalBehavior(counter: Int): Behavior[Request] =
+      Behaviors.receiveMessage[Request] {
+        case Request(replyTo) =>
+          val newCounter = counter + 1
+          replyTo ! newCounter
+          internalBehavior(newCounter)
+      }
+
+    internalBehavior(0)
+  }
+
+  /**
+   * This actor sends a ask to 'askMeRef' at bootstrap and forward the answer to the probe.
+   * We will use it through out this test.
+   */
+  def behavior(askMeRef: ActorRef[Request], probe: TestProbe[Int]) = Behaviors.setup[Int] { context =>
+    implicit val timeout: Timeout = 1.second
+
+    // send a message to interactWithRef
+    context.ask(askMeRef, Request) {
+      case Success(res) => res
+      case Failure(ex)  => throw ex
+    }
+
+    def internalBehavior(): Behavior[Int] = {
+      Behaviors.receiveMessage { num =>
+        // receive response from interactWithRef and sent to prob
+        probe.ref ! num
+        internalBehavior()
+      }
+    }
+
+    internalBehavior()
+  }
+
+  "IgnoreActorRef instance" should {
+
+    "ignore all incoming messages" in {
+
+      val askMeRef = testKit.spawn(askMeActorBehavior)
+
+      val probe = testKit.createTestProbe[Int]("response-probe")
+      askMeRef ! Request(probe.ref)
+      probe.expectMessage(1)
+
+      // this is more a compile-time proof
+      // since the reply is ignored, we can't check that a message was sent to it
+      askMeRef ! Request(testKit.system.ignoreRef)
+      probe.expectNoMessage()
+
+      // but we do check that the counter has increased when we used the ActorRef.ignore
+      askMeRef ! Request(probe.ref)
+      probe.expectMessage(3)
+    }
+
+    // this is kind of obvious, the Future won't complete because the ignoreRef is used
+    "make a Future timeout when used in a 'ask'" in {
+
+      implicit val timeout: Timeout = 500.millis
+      implicit val scheduler = testKit.system.scheduler
+      val askMeRef = testKit.spawn(askMeActorBehavior)
+
+      eventually {
```

Comment:
The `eventually` is in the wrong place here. It should only wrap the last line (`failure shouldBe a[TimeoutException]`). 

That future should timeout because we pass the ignoreRef to the command